test/els_execute_command_SUITE.erl

```diff
@@ -0,0 +1,100 @@
+-module(els_execute_command_SUITE).
+
+%% CT Callbacks
+-export([ suite/0
+        , init_per_suite/1
+        , end_per_suite/1
+        , init_per_testcase/2
+        , end_per_testcase/2
+        , groups/0
+        , all/0
+        ]).
+
+%% Test cases
+-export([ erlang_ls_info/1
+        , strip_server_prefix/1
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include_lib("common_test/include/ct.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+%%==============================================================================
+%% Types
+%%==============================================================================
+-type config() :: [{atom(), any()}].
+
+%%==============================================================================
+%% CT Callbacks
+%%==============================================================================
+-spec suite() -> [tuple()].
+suite() ->
+  [{timetrap, {seconds, 30}}].
+
+-spec all() -> [{group, atom()}].
+all() ->
+  [{group, tcp}, {group, stdio}].
+
+-spec groups() -> [atom()].
+groups() ->
+  els_test_utils:groups(?MODULE).
+
+-spec init_per_suite(config()) -> config().
+init_per_suite(Config) ->
+  els_test_utils:init_per_suite(Config).
+
+-spec end_per_suite(config()) -> ok.
+end_per_suite(Config) ->
+  els_test_utils:end_per_suite(Config).
+
+-spec init_per_testcase(atom(), config()) -> config().
+init_per_testcase(TestCase, Config) ->
+  els_test_utils:init_per_testcase(TestCase, Config).
+
+-spec end_per_testcase(atom(), config()) -> ok.
+end_per_testcase(TestCase, Config) ->
+  els_test_utils:end_per_testcase(TestCase, Config).
+
+%%==============================================================================
+%% Testcases
+%%==============================================================================
+
+-spec erlang_ls_info(config()) -> ok.
+erlang_ls_info(Config) ->
+  Uri = ?config(code_navigation_uri, Config),
+  PrefixedCommand
+    = els_execute_command_provider:add_server_prefix(<<"info">>),
+  #{result := Result}
+    = els_client:workspace_executecommand(PrefixedCommand, [#{uri => Uri}]),
+  Expected = [],
+  ?assertEqual(Expected, Result),
+  CheckFun = fun() -> case els_client:get_notifications() of
+                        [] -> false;
+                        Notifications -> {true, Notifications}
+                      end
+             end,
+  {ok, [Notification]} = els_test_utils:wait_for_fun(CheckFun, 10, 3),
```

Comment:
It looks like notifications are only introduced for testing purposes, to understand when something was called. I think this could be achieved differently, by mocking, passing-through and message passing. I can show you an example, if you need.

Reply:
Well, the core functionality of the feature being added is to send that notification.  And the client is intended to be an interface to the server. So there *should* be a way to actually receive the notifications, by something using the client.

If you feel strongly and can show an alternative based on mocking I will change it, but not convinced it is the right approach.

Reply:
I will not push for this, I think it can eventually be cleaned up later. From my perspective, the only blocker for this PR is to have a configurable list of code lenses for the project via the config file (empty by default for now, until we have a bunch). But I am also ok for merging this and addressing that in a follow-up before next release. @jfacorro opinions?

Reply:
Ok. And given it is a feature for debugging config files, I would suggest we make it a configuration parameter from the client rather, since we may not be adding the feature to the config file we expect to be used.